src/core/Akka.Remote/EndpointManager.cs

```diff
@@ -404,7 +404,7 @@ protected override SupervisorStrategy SupervisorStrategy()
                     {
                         if (hopeless.Uid.HasValue)
                         {
-                            _log.Error("Association to [{0}] with UID [{1}] is irrecoverably failed. Quarantining address.",
+                            _log.Error(hopeless.InnerException, "Association to [{0}] with UID [{1}] is irrecoverably failed. Quarantining address.",
```

Comment:
We weren't properly propagating the internal exceptions thrown during `HopelessAssociation` events earlier; this fixes that.